velox/exec/PartitionFunction.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <velox/exec/HashPartitionFunction.h>
+#include <velox/exec/RoundRobinPartitionFunction.h>
+#include "velox/core/PlanNode.h"
+
+namespace facebook::velox::exec {
+
+void registerPartitionFunctionSerDe() {
+  auto& registry = DeserializationWithContextRegistryForSharedPtr();
+
+  registry.Register(
+      "HashPartitionFunctionSpec", HashPartitionFunctionSpec::deserialize);
+  registry.Register(
+      "RoundRobinPartitionFunctionSpec",
+      RoundRobinPartitionFunctionSpec::deserialize);
+  registry.Register(
```

Comment:
Do we need to remove the register in PlanNode::registerSerDe()? @mbasmanova Thanks!

Reply:
PlanNode::registerSerDe registers GatherPartitionFunctionSpec, hence, we cannot just remove it. We can move GatherPartitionFunctionSpec out of PlanNode.h into its own file like RoundRobin and Hash specs, then add serde here.

Reply:
GatherPartitionFunctionSpec still need to stay in core namespace as it is directly used by PlanNode.h/cpp. Thanks!